### DISCUSSION TITLE
Removal of the feed

### DISCUSSION THREAD

**Discussion by jonjomckay** [jonjomckay/fritter#699] [General]
**TLDR; I'm not going to be reinstating the feed, as it's too hard to maintain and bad for people's mental health.**

Hi all. I won't be reinstating the feed to Fritter. I've been thinking about it a lot this past week, and there are multiple reasons why I don't think it's a good idea. 

As I mentioned above, maintenance of the feed is by far the largest time and effort drain when developing Fritter. It's the least fun and the least interesting part of the app for me to work on. It's fragile (out of necessity because there's no direct "give me the feed" API on Twitter), and prone to a lot of breakage, due to Twitter constantly changing their APIs. The endpoint being completely removed recently, then being added back in was the final straw here - I simply don't want to, and can't, maintain something that relies on something so unreliable. I don't like it, and I know you all don't like it when you get cryptic error messages for days on end - it's almost never because of me, it's Twitter breaking things!

Alongside that, removing the feed was a bit of a cathartic moment for me. Since I took it out, I'm able to think about Fritter without feeling frustrated or overwhelmed. I don't mean this in a negative way, but some of the comments on recent issues and Play Store reviews are starting to bring those feelings back, and I'd rather not have that looming over my head.

In my head, Fritter was never intended to be another one of those endless scroll social media apps. I've never been a fan of doomscrolling, and the mental health issues it can lead to or exacerbate are not something that I would like to encourage. GitHub issue peer pressure is the true killer 🙂

Fritter started off as a fun project, to create a privacy friendly viewer for Twitter (well, Nitter, technically), and I'd like to take it back to that - something fun and useful for me to create, and for the user to enjoy.

For those who disagree with my decision, or don't respect it, I would really encourage you to research the mental dangers of doomscrolling and endless social media feeds. I know a lot of you have it under control, and others don't want to be told what to do, but I would really encourage and appreciate it. There's a very good accessible article [from the BBC on it here](https://www.bbc.com/worklife/article/20210226-the-darkly-soothing-compulsion-of-doomscrolling).

@TheHCJ if you'd like to continue a fork that has the feed, I'm happy for you to do so (MIT license, bro), but we should chat about your branding, changing the use of the Fritter name, choosing a new package name, etc. It would be nice to share a base, as it doesn't pay to write the same things twice 😅

Thank you all for understanding.

**Comment by ghost** [comment]
Thank you for the update! 😀

**Reply by ghost** [reply]
I have now started my fork of Fritter called Quacker! You can check it out here: https://github.com/TheHCJ/Quacker

**Comment by summersab** [comment]
Would you be willing to implement something like this suggestion (opening the topic/hashtag as a web search intent)?
https://github.com/jonjomckay/fritter/issues/668#issuecomment-1530822234

Personally, I agree with you about social media. I liked using Fritter because a) I didn't need to sign in,  and b) it truncated the number of posts available in the feed. I basically used it once or twice a week to see what the "news" was (or, more accurately, why people were "outraged" today).

Since my regular news sources aren't social media, it was interesting to check in with the nonsense now and then. Being able to launch a web search by tapping the hashtags would let me at least figure out why something is "trending." I don't need (or really want) to read what people are saying about the topic. This would be a nice middle-ground.

**Reply by ghost** [reply]
Tweet searching does work, it's a matter of time until @jonjomckay allows it to happen

**Comment by geominorai** [comment]
Doomscrolling depends on the content the user subscribes to, which is ultimately based on their choice and tendency that is beyond anyone else's control.

Like the above comment, some of us check for news only periodically as a summary on world events. In these cases, we would likely be following proper news outlets rather than some rehashed outrage/clickbait reaction media like on Facebook.  There is also use in following important local news alerts like for weather, earthquakes, and public safety that are only practical if subscribed together in bulk rather than viewed individually one by one. 

There are other uses for the feed like following artists and funny/motivational posters that is the opposite side to doomscrolling that is now no longer possible with the feed removal. It instead becomes tedious, demotivating, and punishing to have to open all the profiles individually.  Please do not let one negative aspect of media consumption overshadow the other important uses, especially if it is the result of individual choice.

**Reply by geominorai** [reply]
As a side note, Fritter may have been the last reliable privacy-oriented news source feed for me, as Reddit is about to kill third-party apps by charging for API access as well, and even if you sacrifice privacy by paying a subscription fee, it may not even grant full access without using the official app with all its trackers.

**Comment by geominorai** [comment]
If maintenance of the feed endpoint is the major significant stressor, please consider my proposal for aggregating feeds by distributing the load over time as described in https://github.com/jonjomckay/fritter/issues/668#issuecomment-1528862185

Since it would not depend on any extra endpoint, as long as the mechanism for loading individual profile pages still works (which the current simplified version of the app does), it would basically stay working maintenance-free.

**Comment by GeographicCone** [comment]
@jonjomckay I understand your decision is final. You're of course free to decide what to do in your free time. I appreciate what you've been doing so far. That being said, I hope there's a fork that retains this functionality.

I use the app to follow current events. I subscribe to a lot of accounts, grouped by topic. Without a feed, to check, for example, what's going on in Ukraine right now, I'd have to tap back and forth 50+ times between individual profiles. It's completely impractical. Once the latest version contributed by @TheHCJ (thank you!) stops working, which it eventually will, I'll just have to find another way to access the content I want that doesn't include your app. Of course this is just my use case.

I wanted to comment on your use of this word specifically:

> doomscrolling

As I understand it, this refers to the phenomenon where the feed is algorithmic, full of "featured" or "curated" content, and clickbait suggestions "you'll also like," taking you down the rabbit hole. Fritter's feed is chronological, and it only shows what the user wants it to show. It puts the user in charge, which also comes with responsibility.

The fact that some people might make wrong choices, doesn't mean everybody should be prevented from having a choice at all. Otherwise you might as well make an argument against releasing any apps anymore, since the way many (if not most) people use smartphones is generally unhealthy for them. While we're at it, why not also restrict the sale of liquor? The same argument can be made that it causes a lot of problems too.

People's feeds are a reflection of their choices. You linked to an article from the BBC. The "legacy" media tend to focus on all the alarmist aspects of "social media" while remaining oblivious to the issues in their own garden. I've just opened the BBC homepage, and with stories like "The world's oldest hamburger" or "The 17 most eye-catching looks of [some people I neither know or care about]," I'd rather have my Fritter feed anytime instead.

I'm not piling on the BBC, it's a reliable source on what it reports. But the scope of any single outlet's coverage is a reflection of their policies, as nobody has infinite resources. For example, for the BBC, Asia = India. I'm in Taiwan right now, and I don't know what it would take for them to write an article about Taiwan. Maybe it'd help if a member of the royal family mentioned it... on their social media? Which brings me to my second point.

It's also that the traditional news (which I also read) are often not the first to report on events anymore. A lot of the news reporting these days is just regurgitating what was already posted on Twitter earlier. Many of the accounts I follow are journalists, who often post snippets of their upcoming stories before the full article is ready.

Without digressing any further off-topic (apologies for that), I think it's ultimately a game of "pick your poison." Fritter put me in charge of this decision, and I really appreciate your work on it.

For the record, in case someone brings it up, I'm not talking about what is or isn't good for children here, only adults. Children have parents, and if parenting is outsourced to the app store, then that's an entirely different problem, which should not be addressed at the app level.

One last note: you could easily implement a feature in the feed that limits scrolling past a configurable # of posts, if the user so chooses. Say, show a message in the feed if the user has already seen 50 posts (soft limit), and do not scroll past 100 posts at all (hard limit). You could also implement a timer in the app to warn the user if they spent say over 30 minutes scrolling through.

The details could be worked out, but there exists a technological solution to the problem you describe that doesn't involve taking away the feature in its entirety. Of course I understand you have other reasons (including personal) you don't want to work on this anymore, and I respect your decision.

**Comment by ghost** [comment]
Just archive the project then. Tell people to go to Quacker in your README.
Only a very tiny amount of users would prefer whatever this is, you consider this a feature but we consider this a bug.
Just because you hate the concept of feeds, that doesn't mean we don't find it convenient.
